### DISCUSSION TITLE
How to install into PlatfromIO?

### DISCUSSION THREAD

**Discussion by zebra1993** [SpenceKonde/megaTinyCore#551] [Library discussion]
Im currently trying to import the MegaTinyCore into PlatformIO. 

Ive checked [Here ](https://github.com/SpenceKonde/megaTinyCore/blob/master/megaavr/extras/PlatformIO.md) to see if there is a guide on how to set it up.
The Template provided doesnt work when i change the Board to Attiny1624.

Copiing the the 2.4.2 Release into the packagesfolder and naming it "framework-megaTinyCore" didnt make it show up as a selectable platform.

I have added a file for the Attiny1624 in the atmelmegaavr Platform (just copy pasted an existing board that uses megatinycore)
but i get lots of errors about timers and other missing/wrong things.

Will i have to make a custom framework/Platform to get it running? Which would be somewhat above my head, could try though.

Alternativly i think ArduinoIDE should work aswell, im just used it PIO as all the other projects i have are also there. 

Ideas/ suggestions reguarding the Installation?

I have a handfull of ATtiny1624 that i want to use for a CAN - SPI - Tiny - I2C project.

**Comment by Gobol** [comment]
2.4.2 isn't pushed into platformio's, yet. 
Two ways:
- use Arduino "IDE"
- wait.

**Comment by SpenceKonde** [comment]
Wouldn't recommend the waiting. I'm sure that somehow it is possible to make it work, I just don't know enough about pio to advise. Are you able to get any soft of debugginginfo

**Reply by Gobol** [reply]
Platformio relies on json’s defining whole toolchain and available target boards. I think it should be possible to use 2.4.2 after replace, but still one must write feature=2 targets definitions. Without it, you can rely on actually available targets.

**Comment by Gobol** [comment]
Also, it is very important nowadays to get proper toolchain. ATtiny 32k is last option left, because of shortages…

**Comment by SpenceKonde** [comment]
Also, that platformio document is pretty much obsolete by now, and I don't know enough about platformio to know how to update it. 

Like - there are a whole bunch of new clock options for internal tuned oscillator, 

What I don't understand about PIO is how much of the platform.txt and boards.txt it uses, and how much has to be specified in the pio config, and attachInterrupt has a new implementation that sidesteps it's habit of grabbing onto every possible pin interrupt vector, and reduces it's flash footprint by a couple hundred bytes under most circumstances. The printf option now has a menu. Wire needs the wire mode specified. 

I don't know whether I can expect *any* of the stuff I enter in boards.txt and platform.txt to get carried over, though. And if it doesn't, 2.5.0 is going to totally hose PIO, since there's a bunch of stuff that relies on the default menu selections doing something. Is that a problem? I don't really know...

**Comment by Gobol** [comment]
Wanting to have more sophisticated IDE I tried to move to VisualMicro and Atmel/MCHPStudio - looks very good and integrates with arduino almost seamlessly. It is not free, though…
